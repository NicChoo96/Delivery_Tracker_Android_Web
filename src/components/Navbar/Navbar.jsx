import React, { useEffect, useState, Suspense } from 'react';
import { Paper, Tabs, Tab, Button } from '@material-ui/core';
import { withFirebase } from '../../Firebase';
import { history } from '../../utils';
import Loader from '../Loader/Loader';

import './Navbar.scss';

/**
 * Renders a suggested route based on an array of coordinates
 * @category UI Component
 * @subcategory General
 * @class NarBar
 * @param {Firebase} firebase CRUD interface to communiate with firebase backend service
 * @param {number} tabValue position of the page in the tab menu
 * @param {ReactComponent} react component to render
 * @return {ReactComponent}
 */
const NarBar = ({ firebase, tabValue = 0, Page }) => {
  const [disabled, setDisabled] = useState(true);

  const logOutHandler = (event) => {
    event.preventDefault();
    firebase.signOut();
    console.log('user signed out');
    history.push('/signin');
  };

  // woo using currying for the first time in ma life ever
  const redirectHandler = (url) => (event) => {
    event.preventDefault();
    history.push(url);
  };

  useEffect(() => {
    if (firebase === null) return;

    firebase.auth.onAuthStateChanged((userAuth) => {
      setDisabled(userAuth ? false : true);
    });
  });

  return (
    <>
      <Paper square className="navbar__wrapper">
        <span className="navbar__app-label">
          <img
            className="navbar__app-icon"
            alt="trackee icon"
            src={process.env.PUBLIC_URL + '/image/headerLogo.png'}
          />
        </span>
        <Tabs value={tabValue} indicatorColor="primary" textColor="primary">
          <Tab
            disabled={disabled}
            label="Home"
            className="navbar__nav-link"
            onClick={redirectHandler('/jobs')}
          />
          <Tab
            disabled={disabled}
            label="Profile"
            className="navbar__nav-link"
            onClick={redirectHandler('/profile')}
          />
        </Tabs>
        <span className="navbar__signout">
          <Button disabled={disabled} onClick={logOutHandler}>
            Logout
          </Button>
        </span>
      </Paper>
      <Suspense fallback={Loader()}>
        <div>{Page}</div>
      </Suspense>
    </>
  );
};

export default withFirebase(NarBar);
