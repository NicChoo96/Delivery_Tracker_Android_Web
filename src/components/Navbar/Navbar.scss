.navbar__wrapper {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  box-sizing: inherit;
  line-height: 24px;

  position: fixed;;
  top: 0;
  width: 100%;
  z-index: 1000 ! important;
}

.navbar__app-label {
  margin: auto 1em;
  font-size: 1.2em;
}

.navbar__app-icon {
  height: 2em;
}

.navbar__signout {
  display: inline-flex;
  margin: auto 2em;
}
