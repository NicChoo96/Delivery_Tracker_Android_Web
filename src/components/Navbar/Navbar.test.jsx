import React, { lazy } from 'react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';

import {
  cleanup,
  render,
  waitForElement,
  // getByTestId,
  // wait,
} from '@testing-library/react';
import renderer from 'react-test-renderer';

import Navbar from './Navbar';

const MockComponent = lazy(() => import('../LazyComponentMock'));

afterEach(cleanup);

describe('Navbar', () => {
  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Navbar />, div);
  });

  test('render with page', async () => {
    const { getByTestId } = render(
      <Navbar tabValue={0} Page={<MockComponent />} />
    );
    const lazyElement = await waitForElement(() =>
      getByTestId(`navbar__lazy-component`)
    );
    expect(lazyElement).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Navbar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
