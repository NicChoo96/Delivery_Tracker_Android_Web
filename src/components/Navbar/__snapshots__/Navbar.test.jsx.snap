// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches snapshot 1`] = `
Array [
  <div
    className="MuiPaper-root navbar__wrapper MuiPaper-elevation1"
  >
    <span
      className="navbar__app-label"
    >
      <img
        alt="trackee icon"
        className="navbar__app-icon"
        src="/image/headerLogo.png"
      />
    </span>
    <div
      className="MuiTabs-root"
    >
      <div
        className="MuiTabs-scroller MuiTabs-fixed"
        onScroll={[Function]}
        style={
          Object {
            "marginBottom": null,
            "overflow": "hidden",
          }
        }
      >
        <div
          className="MuiTabs-flexContainer"
          onKeyDown={[Function]}
          role="tablist"
        >
          <button
            aria-selected={true}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary navbar__nav-link Mui-disabled Mui-selected Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Home
            </span>
            <span
              className="PrivateTabIndicator-root-1 PrivateTabIndicator-colorPrimary-2 MuiTabs-indicator"
              style={Object {}}
            />
          </button>
          <button
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary navbar__nav-link Mui-disabled Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Profile
            </span>
          </button>
        </div>
      </div>
    </div>
    <span
      className="navbar__signout"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Logout
        </span>
      </button>
    </span>
  </div>,
  <div />,
]
`;
